src/main/java/org/sagebionetworks/web/client/widget/table/v2/TableEntityWidget.java

```diff
@@ -195,37 +243,63 @@ private void configureActions() {
 		this.actionMenu.setActionListener(Action.UPLOAD_TABLE_DATA, action -> {
 			onUploadTableData();
 		});
-		this.actionMenu.setActionListener(Action.DOWNLOAD_TABLE_QUERY_RESULTS, action -> {
-			onDownloadResults();
-		});
 		this.actionMenu.setActionListener(Action.EDIT_TABLE_DATA, action -> {
 			onEditResults();
 		});
 		this.actionMenu.setActionListener(Action.SHOW_TABLE_SCHEMA, action -> {
-			isShowingSchema = !isShowingSchema;
-			view.setSchemaVisible(isShowingSchema);
-			String showHide = isShowingSchema ? HIDE : SHOW;
+			boolean isVisible = !view.isSchemaVisible();
+			view.setSchemaVisible(isVisible);
+			String showHide = isVisible ? HIDE : SHOW;
 			actionMenu.setActionText(Action.SHOW_TABLE_SCHEMA, showHide + entityTypeDisplay + SCHEMA);
 		});
 
 		this.actionMenu.setActionListener(Action.SHOW_VIEW_SCOPE, action -> {
-			isShowingScope = !isShowingScope;
-			view.setScopeVisible(isShowingScope);
-			String showHide = isShowingScope ? HIDE : SHOW;
+			boolean isVisible = !view.isScopeVisible();
+			view.setScopeVisible(isVisible);
+			String showHide = isVisible ? HIDE : SHOW;
 			actionMenu.setActionText(Action.SHOW_VIEW_SCOPE, showHide + SCOPE + entityTypeDisplay);
 		});
 
 		this.actionMenu.setActionListener(Action.EDIT_DATASET_ITEMS, action -> {
-			if (!isShowingItemsEditor) {
-				actionMenu.setActionVisible(Action.EDIT_DATASET_ITEMS, false);
-				view.setItemsEditorVisible(true);
-				isShowingItemsEditor = true;
-				view.setQueryResultsVisible(false);
-			};
+			showDatasetItemsEditor();
 		});
 
+		// Edit data
+		this.actionMenu.setActionEnabled(Action.EDIT_TABLE_DATA, hasQueryableData);
+		if (hasQueryableData) {
+			this.actionMenu.setEditTableDataTooltipText("Bulk edit cell values");
+		} else {
+			this.actionMenu.setEditTableDataTooltipText("There is no data to edit");
+		}

```

Comment:
I think this addresses [SWC-5852](https://sagebionetworks.jira.com/browse/SWC-5852)